examples/integration/tiles/tile_smoothing.py

```diff
@@ -14,9 +14,10 @@
 p2.y_range = p1.y_range
 
 # Add smoothed tiles (the default)
-r1 = p1.add_tile(CARTODBPOSITRON)
+tile_provider = get_provider(Provider.CARTODBPOSITRON)
```

Comment:
Yeah with the current version, it could be done as simply as
`CARTODBPOSITRON = Provider.CARTODBPOSITRON` once the deprecation has happened,
and then it should work. I personally like the enum more than strings, because it prevents miss-spelling and the like, but that could also stay undocumented until 2.0 I guess.

Reply:
I think preserving the deprecation warning is more important, we should be as rigorous as possible about telegraphing breaking changes that will occur in the future

Reply:
It is preserved. I just mean when you use `get_provider(CARTODBPOSITRON)` (not the string version), you will receive a deprecation warning even though this exact same behavior will work in 2.0, when the properties are changed to constants. This is due to the fact that the import of CARTODBPOSITRON triggers the warning already

Reply:
Ah, I see. I would say that's OK then, as long as we don't demonstrate that usage anywhere right now, since the converse is also true: normal usage of supported features should not raise deprecations. We can document and convert examples to this style at 2.0

Reply:
Would you remove it from the test case also (due to "demonstrated usage")?

Reply:
I think it's OK to leave the test, it's the docs/examples that I think are relevant here (most users don't read tests AFAIK)